infra/build/functions/target_experiment.py

```diff
@@ -220,11 +220,17 @@ def run_experiment(project_name, target_name, args, output_path, errlog_path,
       f'COVERAGE_EXTRA_ARGS={project.coverage_extra_args.strip()}',
   ])
   steps.append({
-      'name': build_lib.get_runner_image_name(''),
-      'env': env,
+      'name':
+          build_lib.get_runner_image_name(''),
+      'env':
+          env,
+      'entrypoint':
+          '/bin/bash',
       'args': [
-          'coverage',
-          target_name,
+          '-c',
+          (f'timeout 30m coverage {target_name}; ret=$?; [ $ret -eq 124 ] '
```

Comment:
Can you add a comment for where 124 comes from?

Reply:
Done